### ISSUE TITLE
Possible logic error in basis.py: context assignment for differentiable flag

### ISSUE THREAD

**Issue by yuzimu1** [lucidrains/se3-transformer-pytorch#21]
Hi, thanks for the great work!

I was reading through the get_basis function in pytorch/se3_transformer_pytorch/basis.py
, and I noticed this line:

`context = null_context if not differentiable else torch.no_grad`

Shouldn't this be the other way around?

If differentiable == True, we want gradients to flow (i.e., no torch.no_grad()), so context should be null_context.

If differentiable == False, we want to disable gradients, so context should be torch.no_grad.

In that case, I believe the line should be:

context = torch.no_grad if not differentiable else null_context

Could you please confirm if this is intentional or might be a small mistake?

Thanks!

**Comment by lucidrains** [comment]
@yuzimu1 oh no you are correct

thanks for catching this!
